cudax/include/cuda/experimental/__stf/internal/algorithm.cuh

```diff
@@ -0,0 +1,396 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDASTF in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2022-2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+/** @file
+ *
+ * @brief Algorithm construct, to embed a CUDA graph within a task
+ */
+
+#pragma once
+
+#include <cuda/experimental/__stf/allocators/pooled_allocator.cuh>
+#include <cuda/experimental/__stf/internal/context.cuh>
+
+#include <map>
+
+namespace cuda::experimental::stf
+{
+
+/**
+ * @brief Algorithms are a mechanism to implement reusable task sequences implemented by the means of CUDA graphs nested
+ * within a task.
+ *
+ * The underlying CUDA graphs are cached so that they are temptatively reused
+ * when the algorithm is run again. Nested algorithms are internally implemented as child graphs.
+ */
+class algorithm
```

Comment:
I think that's an overly general name... it really carries no information. How about `task_sequence`?